Grid.js

```diff
@@ -0,0 +1,168 @@
+        }
+        for (let i = 0; i < this.numMines;) {
+            let row = getRandomInt(this.gridSize);
+            let col = getRandomInt(this.gridSize);
+            let cell = this.getCell(row, col);
+            if (!cell.isMine) {
+                cell.isMine = true;
+                i++;
+            }
+        }
+    };
+
+    applyFnToNeighbors(row, col, callback){
+        let topLeft = {row: row-1, col: col-1};
+        let topCenter = {row: row-1, col: col};
+        let topRight = {row: row-1, col: col+1};
+        
+        let centerLeft = {row: row, col: col-1};
+        let centerRight = {row: row, col: col+1};
+    
+        let bottomLeft = {row: row+1, col: col-1};
+        let bottomCenter = {row: row+1, col: col};
+        let bottomRight = {row: row+1, col: col+1};
+    
+        let surroundingCells = [topLeft, topCenter, topRight, 
+                                centerLeft, centerRight, 
+                                bottomLeft, bottomCenter, bottomRight];
+        //console.log(surroundingCells);
+        
+        surroundingCells.forEach(c => callback(c));
+    };
+
+    isInBounds(coordinate){
+        return coordinate >= 0 && coordinate < this.gridSize;
+    };
+
+    checkForMine(surroundingCell) {
+        if (!this.isInBounds(surroundingCell.row) || !this.isInBounds(surroundingCell.col)) {
+            return 0;  
+        }
+        return this.getCell(surroundingCell.row, surroundingCell.col).isMine ? 1 : 0;
+    };
+
+    getNumMines(cell, row, col){
+        this.getCell(row,col).numMinesAround += this.checkForMine(cell);
+    };
+
+    findMinesAround(){
+        for (let row = 0; row < this.gridSize; row++) {
+            for (let col = 0; col < this.gridSize; col++) {
+                if (!this.getCell(row, col).isMine){
+                    this.changeCellProperty(row, col, "numMinesAround", 0);
+                    this.applyFnToNeighbors(row, col, (cell) => {this.getNumMines(cell, row, col)});
+                }
+            }
+        };
+    };
+
+    revealNeighbors(cell, revealedCells, zeros){
+                
+        //if its not in Bounds do nothing else
+        if(!this.isInBounds(cell.row) || !this.isInBounds(cell.col)) {
+           return;
+        }
+
+        let currentCell = this.getCell(cell.row, cell.col);
+        // if it's a zero, reveal it and add to zero list
+        // adds it to revealed Cell list, so it can be revealed in display function
+        if (currentCell.numMinesAround == 0 && !currentCell.isRevealed){
+            currentCell.isRevealed = true;
+            zeros.push(currentCell);
+            revealedCells.push(currentCell);
+            // console.log(gridCell);
+        }
+
+        // adds it to revealed cell list and reveals it
+        else if (currentCell.numMinesAround > 0 && !currentCell.isRevealed) {
+            currentCell.isRevealed = true;
+            revealedCells.push(currentCell);
+        }
```

Comment:
style, the comment doesn't add much useful detail imo but leaving it is okay
```suggestion
        } else if (currentCell.numMinesAround > 0 && !currentCell.isRevealed) {
            // adds it to revealed cell list and reveals it
            currentCell.isRevealed = true;
            revealedCells.push(currentCell);
        }
```

Reply:
ideal to not have comments - talked about